tests/plugins/test_generate_maven_metadata.py

```diff
@@ -0,0 +1,259 @@
+    'filename': 'jaxb-core-2.2.11-4-sha1.jar',
+    'group_id': 'org.glassfish.jaxb.sha1',
+    'id': 1269792,
+    'size': 524417,
+    'version': '2.2.11-4'
+}
+
+ARCHIVE_SHA256 = {
+    'artifact_id': 'jaxb-sha256',
+    'build_id': 472397,
+    'checksum': 'ca52bcbda16954c9e83e4c0049277ac77f014ecc16a94ed92bc3203fa13aac7d',
+    'checksum_type': koji.CHECKSUM_TYPES['sha256'],
+    'filename': 'jaxb-core-2.2.11-4-sha256.jar',
+    'group_id': 'org.glassfish.jaxb.sha256',
+    'id': 1269792,
+    'size': 524417,
+    'version': '2.2.11-4'
+}
+
+# To avoid having to actually download archives during testing,
+# the checksum value is based on the mocked download response,
+# which is simply the "filename" and "group_id" values.
+DEFAULT_ARCHIVES = [
+    ARCHIVE_JAXB_SUN_POM,
+    ARCHIVE_JAXB_SUN_JAR,
+    ARCHIVE_JAXB_JAVADOC_SUN_JAR,
+    ARCHIVE_JAXB_GLASSFISH_POM,
+    ARCHIVE_JAXB_GLASSFISH_JAR,
+    ARCHIVE_JAXB_JAVADOC_GLASSFIX_JAR,
+    ARCHIVE_SHA1,
+    ARCHIVE_SHA256,
+]
+
+
+class MockSource(object):
+    def __init__(self, tmpdir):
+        tmpdir = str(tmpdir)
+        self.dockerfile_path = os.path.join(tmpdir, 'Dockerfile')
+        self.path = tmpdir
+
+    def get_build_file_path(self):
+        return self.dockerfile_path, self.path
+
+
+class MockedPathInfo(object):
+    def __init__(self, topdir=None):
+        self.topdir = topdir
+
+    def mavenbuild(self, build):
+        return '{topdir}/packages/{name}/{version}/{release}/maven'.format(topdir=self.topdir,
+                                                                           **build)
+
+    def mavenfile(self, maveninfo):
+        return '{group_id}/{artifact_id}/{version}/{filename}'.format(**maveninfo)
+
+
+def mock_koji_session(koji_proxyuser=None, koji_ssl_certs_dir=None,
+                      koji_krb_principal=None, koji_krb_keytab=None,
+                      build_info=None, archives=None):
+    if not build_info:
+        build_info = DEFAULT_KOJI_BUILD
+    if not archives:
+        archives = DEFAULT_ARCHIVES
+
+    flexmock(koji, PathInfo=MockedPathInfo)
+
+    session = flexmock()
+
+    (flexmock(koji)
+     .should_receive('ClientSession')
+     .once()
+     .and_return(session))
+
+    def mock_get_build(nvr):
+        if nvr == DEFAULT_KOJI_BUILD['nvr']:
+            return DEFAULT_KOJI_BUILD
+        else:
+            return None
+
+    (session
+     .should_receive('getBuild')
+     .replace_with(mock_get_build))
+
+    (session
+     .should_receive('listArchives')
+     .and_return(archives))
+
+    (session
+     .should_receive('krb_login')
+     .and_return(True))
+    return session
+
+
+def mock_fetch_artifacts_by_nvr(tmpdir, contents=None):
+    if contents is None:
+        contents = dedent("""\
+            - nvr: com.sun.xml.bind.mvn-jaxb-parent-2.2.11.4-1
+            """)
+
+    with open(os.path.join(tmpdir, REPO_FETCH_ARTIFACTS_KOJI), 'w') as f:
+        f.write(contents)
+        f.flush()
```

Comment:
Suggest to pass the tmpdir directly and `tmpdir.join(REPO_FETCH_ARTIFACTS_KOJI).write_text(content, 'utf-8')`

Reply:
Done